libraries/botbuilder-dialogs/src/configurable.ts

```diff
@@ -14,28 +16,37 @@ export abstract class Configurable {
      * Fluent method for configuring the object.
      * @param config Configuration settings to apply.
      */
-    public configure(config: object): this {
+    public configure(config: Record<string, unknown>): this {
         for (const key in config) {
-            if (config.hasOwnProperty(key)) {
-                const setting = config[key];
-                if (Array.isArray(setting)) {
-                    this[key] = setting;
-                } else if (typeof setting == 'object') {
-                    if (typeof this[key] == 'object') {
-                        // Apply as a map update
-                        for (const child in setting) {
-                            if (setting.hasOwnProperty(child)) {
-                                this[key][child] = setting[child];
+            if (Object.prototype.hasOwnProperty.call(config, key)) {
+                const setting = config[`${key}`];
+                const converter = this.getConverter(key);
+                if (converter && typeof converter === 'object') {
+                    this[`${key}`] = converter.convert(setting);
+                } else {
+                    if (Array.isArray(setting)) {
+                        this[`${key}`] = setting;
+                    } else if (typeof setting == 'object') {
+                        if (typeof this[`${key}`] == 'object') {
+                            // Apply as a map update
+                            for (const child in setting) {
+                                if (Object.prototype.hasOwnProperty.call(setting, child)) {
+                                    this[`${key}`][`${child}`] = setting[`${child}`];
+                                }
                             }
+                        } else {
+                            this[`${key}`] = setting;
                         }
-                    } else {
-                        this[key] = setting;
+                    } else if (setting !== undefined) {
+                        this[`${key}`] = setting;
                     }
-                } else if (setting !== undefined) {
-                    this[key] = setting;
                 }
             }
         }
         return this;
     }
+
+    public getConverter(_property: string): Converter | ConverterFactory {
```

Comment:
Should this be `abstract` as well to force extension classes to implement the functionality?

Reply:
I was considering make this method abstract, however if we do that, every non-abstract class inherited Dialog must implement this method, many unrelated classes would be touched.

Reply:
Fair enough, good point.